lib/linalg/sparse_vector.cc

```diff
@@ -0,0 +1,57 @@
+/*
+ * Copyright 2021 The Clink Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <iostream>
+
+#include "clink/linalg/sparse_vector.h"
+
+using namespace clink;
+using namespace std;
+
+SparseVector::SparseVector(const int n) { n_ = n; }
+
+double SparseVector::get(int index) {
```

Comment:
It is probably not good to continue execution if `index >= n_`. Users might not check for error in the log if the program continues to run.

We can change the method signature to `llvm::Expected<double> SparseVector::get(int index)` so that the caller knows whether the method has encountered error. Feel free to check TFRT for example usages of llvm::Expected.

Reply:
OK. I'll make the change.